src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1280,7 +1281,62 @@ static void fill_idPtypeNP
             "return false; " +
          "}";
   LM_T(LmtMongo, ("idTtypeNP function: %s", bgP->functionIdPtypeNP.c_str()));
-
+#else
+  orion::BSONObjBuilder outer_obj;
+  orion::BSONObjBuilder anyElementTrue;
+  orion::BSONObjBuilder map_obj;
+  orion::BSONObjBuilder in;
+  orion::BSONArrayBuilder and_arr;
+  orion::BSONObjBuilder and_first;
+  orion::BSONObjBuilder and_second;
+  orion::BSONObjBuilder and_third;
+  orion::BSONObjBuilder and_fourth;
+  orion::BSONObjBuilder regex_obj;
+  orion::BSONArrayBuilder eq_arr;
+  orion::BSONObjBuilder eq_obj;
+  orion::BSONObjBuilder eq_obj_2;
+  orion::BSONObjBuilder eq_obj_3;
+  orion::BSONObjBuilder type_obj;
+  orion::BSONArrayBuilder eq_arr_2;
+  orion::BSONArrayBuilder eq_arr_3;
+  orion::BSONArrayBuilder eq_arr_4;
+  orion::BSONArrayBuilder or_arr;
+
+  map_obj.append("input", "$entities");
+  
+  regex_obj.append("input", "Room1");
+  regex_obj.append("regex", "$$this.id");
+  regex_obj.append("options", "i");
+  and_first.append("$regexMatch", regex_obj.obj());
+  and_arr.append(and_first.obj());
+  
+  eq_arr.append("$$this.isPattern");
+  eq_arr.append("true");
+  and_second.append("$eq", eq_arr.arr());
+  and_arr.append(and_second.obj());
+
+  eq_arr_2.append("$$this.type");
+  eq_arr_2.append("Room");
```

Comment:
This should be hardwired. It depends on the entity type. Have a look to L1274

I think it should be:

```suggestion
  eq_arr_2.append(entityType);
```